DigitalPuzzle/DigitalPuzzle.ino

```diff
@@ -1,349 +1,238 @@
+				if (tagID[3] == EEPROM.read(7)) {
+					gate = 2; // and gate
+					Serial.println("AND GATE");
+				}
+			}
+		}
+	}
+	if (tagID[0] == EEPROM.read(8)) {
+		if (tagID[1] == EEPROM.read(9)) {
+			if (tagID[2] == EEPROM.read(10)) {
+				if (tagID[3] == EEPROM.read(11)) {
+					gate = 3; //or gate
+					Serial.println("OR GATE");
+				}
+			}
+		}
+	}
+	if (tagID[0] == EEPROM.read(12)) {
+		if (tagID[1] == EEPROM.read(13)) {
+			if (tagID[2] == EEPROM.read(14)) {
+				if (tagID[3] == EEPROM.read(15)) {
+					gate = 4; // nand gate
+					Serial.println("NAND GATE");
+				}
+			}
+		}
+	}
+	if (tagID[0] == EEPROM.read(16)) {
+		if (tagID[1] == EEPROM.read(17)) {
+			if (tagID[2] == EEPROM.read(18)) {
+				if (tagID[3] == EEPROM.read(19)) {
+					gate = 5; //nor gate
+					Serial.println("NOR GATE");
+				}
+			}
+		}
+	}
+	if (tagID[0] == EEPROM.read(20)) {
+		if (tagID[1] == EEPROM.read(21)) {
+			if (tagID[2] == EEPROM.read(22)) {
+				if (tagID[3] == EEPROM.read(23)) {
+					gate = 6; //xor gate
+					Serial.println("XOR GATE");
+				}
+			}
+		}
+	}
+	if (tagID[0] == EEPROM.read(24)) {
+		if (tagID[1] == EEPROM.read(25)) {
+			if (tagID[2] == EEPROM.read(26)) {
+				if (tagID[3] == EEPROM.read(27)) {
+					gate = 7; //not gate
+					Serial.println("NOT GATE");
+				}
+			}
+		}
+	}
+	if (tagID[0] == EEPROM.read(28)) {
+		if (tagID[1] == EEPROM.read(29)) {
+			if (tagID[2] == EEPROM.read(30)) {
+				if (tagID[3] == EEPROM.read(31)) {
+					gate = 8; //wire
+					Serial.println("WIRE");
+				}
+			}
+		}
+	}
+	return gate;
 }
 
 
 /*************************************************************
- * Function：ReadSerial 
+ * Function：ReadSerial
  * Description：Reads the incoming serial message and process
-   this. 
+   this.
  * Input parameter：none
  * Return：none
  *************************************************************/
 void ReadSerial()
 {
-    String Q; 
-    String ReadString; 
-
-    while (Serial.available())
-    {
-        delay(10);
-        if (Serial.available() > 0)
-        {
-            char c = Serial.read();
-            if (isControl(c))
-            {
-                break;
-            }
-            ReadString += c;
-        }
-    }
-    Q = ReadString;
-    //-------Checking Serial Read---------
-    if (Q == "CAL") 
-    {
-        CalibrateTags();
-        Serial.println("Calibrating Done!");
-    }
-    else if (Q == "HELP") 
-    {
-        Serial.println("-----------------------------------");
-        Serial.println("---------    Main Menu    ---------");
-        Serial.println("-----------------------------------");
-        Serial.println("CAL             ->   Calibrate Tags");
-        Serial.println("HELP            ->        Main Menu");
-        Serial.println("-----------------------------------");
-    }
-    else 
-    {
-        Serial.println("Unkown input")
-    }
+	String Q;
+	String ReadString;
+
+	while (Serial.available())
+	{
+		delay(10);
+		if (Serial.available() > 0)
```

Comment:
Is dit niet altijd `true`?

Reply:
Not that I know. Voor zover ik weet is "Serial.availabe()>0" alleen true wanneer je data binnenkrijgt